File: submissions/0053-maximum-subarray/solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max=nums[0], temp;
        int i,j;
        for(i=0;i<nums.length;i++)
        {
            temp=0;
            for(j=i;j<nums.length;j++)
            {
                temp=temp+nums[j];
                if(temp>max)
                {
                    max=temp;
                }
            }   
        }
        return max;
    }
}

File: submissions/0053-maximum-subarray/solution.py
class Solution:
    def maxSubArray(self, nums: list[int]) -> int:
        maximum = float('-inf')
        n = len(nums)
        total = 0
        for element in nums:
            total+=element
            if total>maximum:
                maximum = total
            if total<0:
                total = 0
                    
        return maximum
        
